lib/src/component/prop_mixins.dart

```diff
@@ -49,7 +49,7 @@ abstract class _$ReactPropsMixin {
   dynamic _raw$ReactProps$children;
 
   /// The children that were passed in to this component when it was built.
-  List<dynamic> get children {
+  List<dynamic>? get children {
```

Comment:
This being nullable is a bit of a bummer; I'm anxious to see how much of an issue this is in practice though, as we do more testing in consumer repos.

Reply:
We do that only for Dart components; JS components ([here](https://sourcegraph.plat.workiva.net/github.com/Workiva/react-dart/-/blob/lib/react_client/component_factory.dart?L275)) and DOM components ([here](https://sourcegraph.plat.workiva.net/github.com/Workiva/react-dart/-/blob/lib/react_client/component_factory.dart?L296)) don't always have list children. Also, props passed in from JS code (like when using a Dart wrapper in a slot) also may not be lists. See https://github.com/Workiva/over_react/pull/698 for a little more context on that.

The whole "props are always a List" thing in react-dart/over_react has been a bit of a mess for a while, and making it nullable I think moves us in the direction of being closer to how things work in JS land, where children are treated more like opaque content.

Another reason it's nullable is to support partial props maps better (see note [here](https://github.com/Workiva/over_react/commit/5ac47e90cbc3e97f9631cd690063fad57c793c94#diff-533d8fd37b6f0ca0fd1fdd7086da9ddc98bc3b15ac054340ff9ad087159ea2e6R60-R63)). I'm not sure how likely it is that consumers are relying on that behavior, but if making it nullable minimizes the risk for breakages while not being too disruptive, it seems worth it to me.